edu-sync/src/config.rs

```diff
@@ -105,6 +105,32 @@ impl CourseConfigs {
     }
 }
 
+// Expand Tilde to home folder, create the directory if it does not exist and
+// then canonicalize the path.
+pub fn check_dir(path: &PathBuf) -> Result<PathBuf, std::io::Error> {
+    let expanded_path = shellexpand::path::tilde(&path).into_owned();
+    if !expanded_path.try_exists()? {
+        std::fs::create_dir(&expanded_path)?;
+    }
+    std::fs::canonicalize(expanded_path)
+}
+
+// Custom deserializer function to check if the path is absolute and to
+// canonicalize the path
+fn deserialize_absolute_path<'de, D>(deserializer: D) -> Result<PathBuf, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let path_str = String::deserialize(deserializer)?;
+    let path = PathBuf::from(path_str);
+
+    if !path.is_absolute() {
+        return Err(serde::de::Error::custom("Path must be absolute"));
+    }
```

Comment:
This should be removed, right?
If we bail out on a non-absolute path here, there is no point in expanding tilde, since paths starting with tilde are not absolute.

```suggestion
```